packages/menu/src/MenuItem.ts

```diff
@@ -276,6 +324,108 @@ export class MenuItem extends LikeAnchor(Focusable) {
         }
     }
 
+    public closeOverlay?: (leave?: boolean) => Promise<void>;
+
+    protected handleSubmenuClick(): void {
+        this.openOverlay({ immediate: true });
+    }
+
+    protected handlePointerenter(): void {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        } else {
+            this.openOverlay();
+        }
```

Comment:
An option for an early return in case you like it:

```suggestion
        if (this.leaveTimeout) {
            clearTimeout(this.leaveTimeout);
            delete this.leaveTimeout;
            return;
        }
        this.openOverlay();
```